aten/src/ATen/native/cuda/Embedding.cu

```diff
@@ -259,7 +259,9 @@ Tensor embedding_dense_backward_cuda(const Tensor & grad_, const Tensor & indice
   auto grad = grad_.contiguous().view({num_indices, grad_.size(-1)});
   cudaStream_t stream = at::cuda::getCurrentCUDAStream();
 
-  if (num_indices <= 3072 && !scale_grad_by_freq) {
+  static int num_indices_native_kernel = at::cuda::getCurrentDeviceProperties()->major >= 9 ? 4096 : 3072;
```

Comment:
Minor nit: but the wrong heuristic could be chosen if someone ran the same op on two different GPUs, right?

Reply:
Yes, that's correct

Reply:
So is the overhead of this call worth caching it like this?